platform/x11/speechd-so_wrap.h:88:10: fatal error: 'libspeechd.h' file not found

### Godot version

3.x (9be7c49a5d6dda496d6f700b00a9994532be04b6)

### System information

Ubuntu 22.04

### Issue description

When trying to build the 3.x branch I get a warning:
```
Warning: Speech Dispatcher development libraries not found. Disabling Text-to-Speech support.
```
However, the build still fails:
```
platform/x11/speechd-so_wrap.c:86:10: fatal error: 'libspeechd.h' file not found
```
So the build system is aware that I don't have the speech dispatcher development libraries installed, but it still creates the wrapper, and then it is unable to use it.

Related to #63642

### Steps to reproduce

Ensure the speech dispatcher development libraries are not installed.
Try build the Godot 3.x branch on Linux.

### Minimal reproduction project

N/A

--- a/platform/x11/detect.py
+++ b/platform/x11/detect.py
@@ -369,6 +369,7 @@ def configure(env):
             env.Append(CPPDEFINES=["PULSEAUDIO_ENABLED"])
             env.ParseConfig("pkg-config libpulse --cflags")  # Only cflags, we dlopen the library.
         else:
+            env["pulseaudio"] = False
             print("Warning: PulseAudio development libraries not found. Disabling the PulseAudio audio driver.")
 
     if env["speechd"]:
@@ -376,6 +377,7 @@ def configure(env):
             env.Append(CPPDEFINES=["SPEECHD_ENABLED"])
             env.ParseConfig("pkg-config speech-dispatcher --cflags")  # Only cflags, we dlopen the library.
         else:
+            env["speechd"] = False
             print("Warning: Speech Dispatcher development libraries not found. Disabling Text-to-Speech support.")
 
     if platform.system() == "Linux":
@@ -385,6 +387,7 @@ def configure(env):
                 env.Append(CPPDEFINES=["UDEV_ENABLED"])
                 env.ParseConfig("pkg-config libudev --cflags")  # Only cflags, we dlopen the library.
             else:
+                env["udev"] = False
                 print("Warning: libudev development libraries not found. Disabling controller hotplugging support.")
     else:
         env["udev"] = False  # Linux specific

